cluster/node.go

```diff
@@ -12,22 +22,54 @@ type NodeState int
 const (
 	NodeNotReady NodeState = iota
 	NodeReady
+	NodeUnreachable
 )
 
-type Node struct {
-	name    string
-	version string
-	started time.Time
+var client = http.Client{
+	Transport: &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		Proxy:           http.ProxyFromEnvironment,
+		Dial: (&net.Dialer{
+			Timeout:   time.Second * 5,
+			KeepAlive: 30 * time.Second,
+		}).Dial,
+		TLSHandshakeTimeout: time.Second,
+	},
+	Timeout: time.Second,
+}
 
+func NodeStateFromString(s string) NodeState {
+	switch s {
+	case "NodeNotReady":
+		return NodeNotReady
+	case "NodeReady":
+		return NodeReady
+	case "NodeUnreachable":
+		return NodeUnreachable
+	}
+	//default
+	return NodeNotReady
+}
+
+type Node struct {
 	sync.RWMutex
+	name          string
+	version       string
+	started       time.Time
+	remoteAddr    *url.URL
```

Comment:
add a comment here that this will be nil for ThisNode. that's an important detail